chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -158,20 +217,39 @@ def self.build_node_map(node)
             count_map[speed] = count + 1
           end
         end
-
+        # if_remap now contains something like
+        #   {
+        #     "100m1" => "em1",
+        #     "1g1"   => "em1",
+        #     "10g1"  => "em1",
+        #     "100m2" => "em2",
+        #     "1g2"   => "em2",
+        #     "10g2"  => "em2",
+        #     "10m1"  => "em3",
+        #     "100m3" => "em3",
+        #     "1g3"   => "em3",
+        #     "10m2"  => "em4",
+        #     "100m4" => "em4",
+        #     "1g4"   => "em4"
+        #   }
         ans = {}
-        conduits.each do |k,v|
+        conduits.each do |conduit, conduit_data|
           hash = {}
-          v.each do |mk, mv|
-            if mk == "if_list"
-              hash["if_list"] = v["if_list"].map do |if_ref|
-                map_if_ref(if_remap, if_ref)
+          conduit_data.each do |conduit_key, conduit_val|
+            if conduit_key == "if_list"
+              hash["if_list"] = conduit_data["if_list"].map do |if_ref|
+                r = map_if_ref(if_remap, if_ref)
+                if r.nil?
+                  Chef::Log.warn "Unable to find interface for #{node.fqdn} " \
+                                 "matching #{if_ref} out of #{if_remap}"
```

Comment:
> Would it be good enough to change this Chef::Log.warn into a .debug? 

Yeah, I guess that's fine.

> You mean, somewhere like in NetworkService#allocate_ip_by_type?

Yes. And ```NetworkService#enable_interface```. The advantage of doing proper checks there would be that we could actually fail with an error much earlier than during the chef-client run. (Note: I am not say that you should be the one implementing that, it just an idea that I had when reviewing your code)

> That seems to get all its network info from the databag.

Yeah, it would need to be enhanced to call into ```NodeObject#build_node_map```.  Which reminds me that might want to add the debug message there as well.